### PR TITLE
Allow loading HTTPConnectionFactory password from environment variable

### PR DESCRIPTION
**Pull request by fbruton**

### CODE REVIEW

File: src/test/java/emissary/client/HTTPConnectionFactoryTest.java [hunk 1]
```diff
@@ -74,6 +76,20 @@ void loadPWFromFile() throws Exception {
...
     }
 
+    /**
+     * Read a known environment variable configured during setup {@link UnitTest#setupSystemProperties()}
+     *
+     * @throws Exception thrown when an error occurs
+     */
+    @Test
+    void loadPWFromEnv() throws Exception {
+        char[] pw = HTTPConnectionFactory.loadPW("${PROJECT_BASE}");
```
Comment (jpdahlke) [inline src/test/java/emissary/client/HTTPConnectionFactoryTest.java:86]: Should this test both expected patterns?

**Review by jpdahlke — CHANGES_REQUESTED** [review]

File: src/test/java/emissary/client/HTTPConnectionFactoryTest.java (hunk 1 above)
Comment (fbruton) [reply src/test/java/emissary/client/HTTPConnectionFactoryTest.java:86]: > Should this test both expected patterns?

Removed the second pattern. We can update the regex in the future if necessary.

File: src/test/java/emissary/client/HTTPConnectionFactoryTest.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import static emissary.client.HTTPConnectionFactory.CFG_TRUST_STORE_PW;
 import static emissary.client.HTTPConnectionFactory.CFG_TRUST_STORE_TYPE;
 import static emissary.client.HTTPConnectionFactory.DFLT_STORE_TYPE;
+import static org.junit.Assert.fail;
```
Comment (dev-mlb) [inline src/test/java/emissary/client/HTTPConnectionFactoryTest.java:10]: unused import

**Review by dev-mlb — CHANGES_REQUESTED** [review]

**Review by dev-mlb — APPROVED** [review]

File: src/main/java/emissary/client/HTTPConnectionFactory.java [hunk 3]
```diff
@@ -75,6 +77,7 @@ public class HTTPConnectionFactory {
     private static final String HTTP = "http";
     private static final String HTTPS = "https";
     private static final String FILE_PRE = "file://";
+    private final static Pattern ENV_VARIABLE_PATTERN = Pattern.compile("\\$\\{(\\w+)}");
```
Comment (cfkoehler) [inline src/main/java/emissary/client/HTTPConnectionFactory.java:80]: nit: The modifiers should be `static final`
```suggestion
    private static final Pattern ENV_VARIABLE_PATTERN = Pattern.compile("\\$\\{(\\w+)}");
```

**Review by cfkoehler — APPROVED** [review]

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**